### ISSUE TITLE
[C++ verification] problem with the std::enable_if template

### ISSUE THREAD

**Issue by lucasccordeiro** [esbmc/esbmc#1737]
Bug report from an ESBMC user: " I have a problem with the std::enable_if template. Its is defined in <type_traits> and is used extensively around the code I’m trying to compile."

**Comment by lucasccordeiro** [comment]
@XLiZHI : could you please check this issue?

**Comment by lucasccordeiro** [comment]
Example:

````CPP
#include <iostream>
#include <type_traits>

// Function template that takes two arguments and returns their sum
template<typename T>
typename std::enable_if<std::is_integral<T>::value, T>::type
add(T a, T b) {
    return a + b;
}

int main() {
    // Call the add function with integer arguments
    std::cout << "add(3, 4): " << add(3, 4) << std::endl;

    // Uncommenting this line would result in a compile-time error
    //std::cout << "add(3.5, 4.5): " << add(3.5, 4.5) << std::endl;

    return 0;
}
````

ESBMC command line:

esbmc main.cpp

ESBMC output:

````
ESBMC version 7.5.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing alex.cpp
main.cpp:6:15: error: no type named 'enable_if' in namespace 'std'
typename std::enable_if<std::is_integral<T>::value, T>::type
~~~~~~~~~~~~~~^~~~~~~~~
main.cpp:6:24: error: expected unqualified-id
typename std::enable_if<std::is_integral<T>::value, T>::type
                       ^
ERROR: PARSING ERROR
````

**Comment by fbrausse** [comment]
This issue with enable_if has been addressed by #1739 on the current master branch. In case it's not for the user, let's open another issue then.
